### PR TITLE
virtualfish: update test

### PR DESCRIPTION
**Pull request by chenrui333**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

- https://github.com/justinmayer/virtualfish/issues/250

### CODE REVIEW

File: Formula/v/virtualfish.rb [hunk 1]
```diff
@@ -92,7 +92,10 @@ def caveats
     assert_match "new_virtualenv", shell_output('fish -c "vf ls"')
 
     # Delete the virtualenv
-    system "fish", "-c", "vf rm new_virtualenv"
-    refute_path_exists testpath/".virtualenvs/new_virtualenv/pyvenv.cfg"
+    # cannot delete virtualenv on sequoia, upstream bug report, https://github.com/justinmayer/virtualfish/issues/250
+    if OS.mac? && MacOS.version < :sequoia
+      system "fish", "-c", "vf rm new_virtualenv"
+      refute_path_exists testpath/".virtualenvs/new_virtualenv/pyvenv.cfg"
+    end
```
Comment (carlocab) [inline Formula/v/virtualfish.rb:99]:
```suggestion
    return if OS.mac? && MacOS.version >= :sequoia

    system "fish", "-c", "vf rm new_virtualenv"
    refute_path_exists testpath/".virtualenvs/new_virtualenv/pyvenv.cfg"
```
Plus adjust the comment placement, probably

**Comment by carlocab** [comment]
Plus the error seems like a Homebrew sandbox issue rather than something upstream can do anything about

**Comment by chenrui333** [comment]
could be cc @Bo98

File: Formula/v/virtualfish.rb [hunk 2]
```diff
@@ -92,6 +92,9 @@ def caveats
     assert_match "new_virtualenv", shell_output('fish -c "vf ls"')
 
     # Delete the virtualenv
+    # cannot delete virtualenv on sequoia, upstream bug report, https://github.com/justinmayer/virtualfish/issues/250
+    return if OS.mac? && MacOS.version >= :sequoia
+
```
Comment (carlocab) [inline Formula/v/virtualfish.rb:97]:
```suggestion
    # cannot delete virtualenv on sequoia, upstream bug report, https://github.com/justinmayer/virtualfish/issues/250
    return if OS.mac? && MacOS.version >= :sequoia

    # Delete the virtualenv
```

**Comment by Bo98** [comment]
> lus the error seems like a Homebrew sandbox issue rather than something upstream can do anything about

Sort of.

`trash` didn't exist prior to macOS 15 but now does so the `trash` code path now always gets executed. Trash does work in sandboxes (unless explicitly allowed I guess).

Maybe not necessarily an upstream bug but an upstream enhancement could be to add an option to use the `rm -rf` code path.

**Comment by chenrui333** [comment]
I dont feel like it is caused by the sandbox, I can see the same failure when running in my local sequoia 15.1

**Review by fxcoudert — APPROVED** [review]

**Pull request merged**